doc: start migration guide documentation for Jest

EvHaus opened the pull request:
Adds a note about lack of DOM clean up when globals are off (as per https://github.com/vitest-dev/vitest/issues/682#issuecomment-1030984679), and restores the doc change added in https://github.com/vitest-dev/vitest/pull/467 which was removed.

sheremet-va commented:
Not sure it should be in comparisons. 

Also, your doc change wasn't removed, it was moved to `API` section: https://vitest.dev/api/#vi-mock

EvHaus commented:
@sheremet-va Good idea. I've updated this PR to start a `Migration Guides` area instead and added a some additional issues I ran into while migrating some projects.

Might be good to have a few more folks contribute to this doc before making is live as it's pretty light right now. Thoughts?

EvHaus commented on docs/api/index.md:
```diff
@@ -1100,8 +1100,6 @@ Vitest provides utility functions to help you out through it's **vi** helper. Yo
   - If `__mocks__` folder with file of the same name exist, all imports will return its exports. For example, `vi.mock('axios')` with `<root>/__mocks__/axios.ts` folder will return everything exported from `axios.ts`.
   - If there is no `__mocks__` folder or a file with the same name inside, will call original module and mock it. (For the rules applied, see [algorithm](/guide/mocking#automocking-algorithm).)
 
-  Additionally, unlike Jest, mocked modules in `<root>/__mocks__` are not loaded unless `vi.mock()` is called. If you need them to be mocked in every test, like in Jest, you can mock them inside [`setupFiles`](/config/#setupfiles).
```
Probably better to have this in the migration guide as not everyone looking at the API docs will be coming from Jest.

dammy001 commented on docs/.vitepress/config.ts:
```diff
@@ -127,7 +127,11 @@ export default defineConfig({
-            }
+            },
+            {
+              text: 'Migration Guides',
```
I think this should be guide
